### PR TITLE
feat: new design for modals and allow change buttons positions

### PR DESCRIPTION
**Pull request by suevalov**
# Purpose of PR

Implements all v3 changes for Modal and ModalConfirm

* New paddings
* Change ModalHeader background
* By default, all buttons in ModalConfirm have `small` size
* Added optional ability to align button on the right side
   * The aligning on the right side will be a default behavior in V4

## PR Checklist

- [ ] I have read the relevant `readme.md` file(s)
- [ ] All commits follow our [Git commit message convention](https://github.com/contentful/forma-36/tree/master/packages/forma-36-react-components#commits)
- [ ] Tests are added/updated/not required
- [ ] Tests are passing
- [ ] Storybook stories are added/updated/not required
- [ ] Usage notes are added/updated/not required
- [ ] Has been tested based on [Contentful's browser support](https://www.contentful.com/faq/about-contentful/#which-browsers-does-contentful-support)
- [ ] Doesn't contain any sensitive information

### CODE REVIEW

**Review by denkristoffer — APPROVED** [review]

File: packages/forma-36-react-components/src/components/Modal/ModalConfirm/ModalConfirm.stories.tsx [hunk 1]
```diff
@@ -102,9 +116,9 @@ export function ComplexStory(props: ModalConfirmProps) {
         }}
         {...props}
       >
-        <p>
+        <div>
```
Comment (burakukula) [inline packages/forma-36-react-components/src/components/Modal/ModalConfirm/ModalConfirm.stories.tsx:119]: It's totally minor, but we could have a margin-bottom here I think.

File: packages/forma-36-react-components/src/components/Modal/ModalHeader/ModalHeader.css [hunk 2]
```diff
@@ -8,17 +9,16 @@
   flex-shrink: 0;
   align-items: center;
   /* prettier-ignore */
```
Comment (burakukula) [inline packages/forma-36-react-components/src/components/Modal/ModalHeader/ModalHeader.css:11]: Not related, but this might be not needed anymore?

**Review by burakukula — APPROVED** [review]

File: packages/forma-36-react-components/src/components/Modal/ModalHeader/ModalHeader.css (hunk 2 above)
Comment (suevalov) [reply packages/forma-36-react-components/src/components/Modal/ModalHeader/ModalHeader.css]: removed

File: packages/forma-36-react-components/src/components/Modal/ModalConfirm/ModalConfirm.stories.tsx (hunk 1 above)
Comment (suevalov) [reply packages/forma-36-react-components/src/components/Modal/ModalConfirm/ModalConfirm.stories.tsx]: added Paragraph

**Comment by fabe** [comment]
Not sure if this is part of this, but could we bring the modal action buttons closer together? 8px/spacing-xs between them should be enough, otherwise they float around a bit too much.

**Comment by suevalov** [comment]
@fabe yes, sure we can do this

**Pull request merged**